packages/java-parser/src/parser.js

```diff
@@ -68,7 +68,9 @@ class JavaParser extends Parser {
     blocksStatements.defineRules.call(this, $, t);
     expressions.defineRules.call(this, $, t);
 
+    this.firstForUnaryExpressionNotPlusMinus = [];
     this.performSelfAnalysis();
+    expressions.computeFirstForUnaryExpressionNotPlusMinus.call(this);
```

Comment:
`this.firstForUnaryExpressionNotPlusMinus = expressions.computeFirstForUnaryExpressionNotPlusMinus.call(this);`

Would be clearer in terms of tacking state changes